README.md

```diff
@@ -1,60 +1,349 @@
+    - [winix] (WSL 2 instance on Windows for Work with CUDA acceleration)
+  - **Servers**
+    - [nixai] (Locally hosted AI services)
+    - [nixarr] (Media management and automation)
     - [nixcloud] (Replacing usage of external cloud services)
-    - [nixdev] (Development)
+    - [nixdev] (Development environment)
     - [nixio] (Interface between the web & internal services)
-    - [nixmon] (Monitoring)
+    - [nixmon] (Monitoring and observability)
     - [nixserv] (Nix Cache & Build System)
 
 ## Repository Structure
 
+The repository uses an **automatic discovery system** that scans the filesystem to build configurations:
+
 ```
 .
-├─ home          # Root for all user homes
-├─── shared      #
-├─ hosts         # Root for all hosts
-├─── shared      # Auto-Imported modules for all hosts
-├─── desktop     # Desktop NixOS systems
-├───── shared    # Auto-Imported modules for desktops
-├─── laptop      # Laptop NixOS Systems
-├───── shared    # Auto-Imported modules for laptops
-├─── server      # Server NixOS Systems
-├───── shared    # Auto-Imported modules for servers
-├─ lib           # Extension to the nixpkgs lib
-├─ modules       # Custom NixOS and home-manager modules
-├─ overlays      # NixPkgs overlays
-└─ pkgs          # Custom Packages
+├─ home              # Root for all user homes (auto-discovered)
+│  ├─── racci        # User-specific configurations
+│  ├─── root         # Root user configurations  
+│  └─── shared       # Shared home-manager modules
+├─ hosts             # Root for all hosts (auto-discovered by device type)
+│  ├─── shared       # Auto-imported modules for all hosts
+│  │    ├─── global  # Core system configuration (locale, networking, etc.)
+│  │    └─── optional# Optional modules for specific use cases
+│  ├─── desktop      # Desktop NixOS systems
+│  │    ├─── shared  # Auto-imported modules for desktops
+│  │    ├─── nixmi   # Personal desktop with hardware acceleration
+│  │    └─── winix   # WSL2 development environment
+│  ├─── laptop       # Laptop NixOS Systems  
+│  │    └─── shared  # Auto-imported modules for laptops
+│  └─── server       # Server NixOS Systems
+│       ├─── shared  # Auto-imported modules for servers
+│       ├─── nixai   # AI services host
+│       ├─── nixarr  # Media automation host
+│       ├─── nixcloud# Cloud services replacement
+│       ├─── nixdev  # Development services
+│       ├─── nixio   # Web interface and proxy
+│       ├─── nixmon  # Monitoring stack
+│       └─── nixserv # Nix cache and build system
+├─ lib               # Extensions to nixpkgs lib and custom builders
+│  └─── builders     # System and home-manager configuration builders
+├─ modules           # Custom NixOS and home-manager modules
+├─ overlays          # NixPkgs overlays for package modifications
+├─ pkgs              # Custom packages not in nixpkgs
+├─ utils             # Utility scripts for development and debugging
+└─ docs              # Additional documentation
```

Comment:
This shouldn't contain direct references the machines and is only intended as a overview of the generics